bootstrap-salt.sh

```diff
@@ -1244,18 +1244,40 @@ __gather_system_info() {
 
 
 #---  FUNCTION  -------------------------------------------------------------------------------------------------------
-#          NAME:  __get_dpkg_architecture
-#   DESCRIPTION:  Determine primary architecture for packages to install on Debian and derivatives
+#          NAME:  __check_dpkg_architecture
+#   DESCRIPTION:  Determine the primary architecture for packages to install on Debian and derivatives
+#                 and issue all necessary error messages.
 #----------------------------------------------------------------------------------------------------------------------
-__get_dpkg_architecture() {
+__check_dpkg_architecture() {
     if __check_command_exists dpkg; then
         DPKG_ARCHITECTURE="$(dpkg --print-architecture)"
     else
         echoerror "dpkg: command not found."
         return 1
     fi
 
-    return 0
+    __REPO_ARCH="$DPKG_ARCHITECTURE"
+
+    if [ "$DPKG_ARCHITECTURE" = "i386" ]; then
+        echoerror "$_REPO_URL likely doesn't have all required 32-bit packages for Ubuntu $DISTRO_MAJOR_VERSION (yet?)."
+
+        if [ "$ITYPE" != "git" ]; then
+            echoerror "You can try git installation mode, i.e.: sh ${__ScriptName} git v2016.11.5"
+        fi
+
+        # amd64 is just a part of repository URI, 32-bit pkgs are hosted under the same location
+        __REPO_ARCH="amd64"
+
+    elif [ "$DPKG_ARCHITECTURE" != "amd64" ]; then
+        echoerror "$_REPO_URL doesn't have packages for your system architecture: $DPKG_ARCHITECTURE."
+
+        if [ "$DPKG_ARCHITECTURE" != "armhf" ]; then
+            echoerror "Try git installation mode with pip and disable SaltStack apt repository, for example:"
+            echoerror "    sh ${__ScriptName} -r -P git v2016.11.5"
+        elif [ "$ITYPE" != "git" ]; then
+            echoerror "You can try git installation mode, i.e.: sh ${__ScriptName} git v2016.11.5"
+        fi
+    fi
```

Comment:
As I see, this should work properly during stable installation on Debian/Raspbian `armhf`, but will output unnecessary error messages which may confuse users.
How about to simplify this whole block to a `case` statement?